packages/contracts/test/ClaimReportNegRep.test.ts

```diff
@@ -0,0 +1,358 @@
+            reportedEpoch,
+            currentEpoch,
+            currentNonce,
+            chainId,
+            attesterId,
+            type,
+        })
+
+        const { publicSignals, proof } = await genProofAndVerify(
+            circuit,
+            reportNegRepCircuitInputs
+        )
+        const flattenedProof = flattenProof(proof)
+        flattenedProof[0] = BigInt(0)
+
+        await expect(
+            app.claimReportNegRep(
+                publicSignals,
+                flattenedProof,
+                identifier,
+                posterPunishment
+            )
+        ).to.be.reverted
+
+        posterState.stop()
+    })
+    it('should revert with wrong epoch', async () => {
+        const posterState = await genUserState(poster.id, app)
+        const identitySecret = poster.id.secret
+
+        // elapsing 5 epoch
+        await ethers.provider.send('evm_increaseTime', [epochLength * 5])
+        await ethers.provider.send('evm_mine', [])
+
+        const currentNonce = 0
+        const currentEpoch = BigInt(44444) // correct epoch = 20
+
+        const circuit = 'reportNegRepProof'
+
+        const type = 0
+        const reportNegRepCircuitInputs = genReportNegRepCircuitInput({
+            reportedEpochKey,
+            identitySecret,
+            reportedEpoch,
+            currentEpoch,
+            currentNonce,
+            chainId,
+            attesterId,
+            type,
+        })
+
+        const { publicSignals, proof } = await genProofAndVerify(
+            circuit,
+            reportNegRepCircuitInputs
+        )
+
+        const flattenedProof = flattenProof(proof)
+
+        await expect(
+            app.claimReportNegRep(
+                publicSignals,
+                flattenedProof,
+                identifier,
+                posterPunishment
+            )
+        ).to.be.revertedWithCustomError(app, 'InvalidEpoch')
+
+        posterState.stop()
+    })
+
+    it('should revert with wrong attester InvalidAttester', async () => {
+        const posterState = await genUserState(poster.id, app)
+        const identitySecret = poster.id.secret
+
+        // elapsing 5 epoch
+        await ethers.provider.send('evm_increaseTime', [epochLength * 5])
+        await ethers.provider.send('evm_mine', [])
+
+        const currentNonce = 0
+        const currentEpoch = await posterState.sync.loadCurrentEpoch()
+        expect(currentEpoch).to.be.equal(25)
+        const wrongAttester = BigInt(444444)
+
+        const circuit = 'reportNegRepProof'
+
+        const type = 0
+        const reportNegRepCircuitInputs = genReportNegRepCircuitInput({
+            reportedEpochKey,
+            identitySecret,
+            reportedEpoch,
+            currentEpoch,
+            currentNonce,
+            chainId,
+            attesterId: wrongAttester,
+            type,
+        })
+
+        // this will be
+        try {
+            await genProofAndVerify(circuit, reportNegRepCircuitInputs)
```

Comment:
this function only get { isValid, publicSignals, proof }, should call `claimReportNegRep` to get reverted message.
```typescript
await expect(
    app.claimReportPosRep(
        publicSignals,
        flattenedProof,
        identifier,
         posReputation
    )
).to.be.reverted
```

Reply:
This wouldn't correctly get a proof from `genProofAndVerify` :(, so wouldn't get the publicSignals and proof to send in `claimReportPosRep()`